test/messenger/messenger_test.cpp

```diff
@@ -289,7 +289,7 @@ TEST_F(MessengerTests, BasicListenTest) {
 
     // Send "replica1" to the primary to let them know who we are.
     messenger->SendMessage(common::ManagedPointer(&con_primary), "replica1", CallbackFns::Noop,
-                           static_cast<uint8_t>(Messenger::BuiltinCallback::ECHO));
+                           static_cast<uint8_t>(Messenger::BuiltinCallback::NOOP));
```

Comment:
Yes, since the current temp fix at https://github.com/cmu-db/noisepage/pull/1288/files#diff-5c77dbc0c8f92240246000b920bc808448e1e240d6f2475633133dc2e9fd892dR523 will evaluate a message both in `ProcessMsg` and in the customized server callback. 
The original code will not trigger this echo callback at all since it doesn't go through the main loop's callback evaluation. This is just to keep the test behaves the same way (And in fact, without this change, the temp fix above doesn't work) 

It might be better to have your changes to the messenger merged in the master, but this is what I need now to pass tests.